### ISSUE TITLE
jsonb instead of text for json value (copy)

### ISSUE THREAD

**Issue by samuelstroschein** [opral/lix-sdk#46]
### Context

For simplicity reasons we choose `TEXT` as type for JSON columns so that `JSON.stringify()` works instead of something like [https://github.com/kysely-org/kysely/issues/209#issuecomment-1306187965](https://github.com/kysely-org/kysely/issues/209#issuecomment-1306187965)

### Proposal

Implement JSON columns as `JSONB` for efficiency.

**Comment by samuelstroschein** [comment]
The inlang SDK has a rough implementation that could work for lix too [MESDK-173](https://linear.app/opral/issue/MESDK-173/jsonb-instead-of-text-for-json-value). 

If the inlang SDK plugin works for lix as well, we should choose it and close this ticket. As long as we use `BLOB` for the columns, a few bugs here and there with parsing/serialization are OK because they don't lead to a breaking change.

**Comment by martin-lysk** [comment]
Referencing this for better visibility [https://github.com/opral/monorepo/pull/3103/files](https://github.com/opral/monorepo/pull/3103/files)

**Comment by martin-lysk** [comment]
Just pushed a working solution.<br>Some concern about the current approach:

* Hashing of jsonB might not be stable:

> JSONB is not intended as an external format to be used by applications. JSONB is designed for internal use by SQLite only. Programmers do not need to understand the JSONB format in order to use it effectively. Applications should access JSONB only through the [JSON SQL functions](https://sqlite.org/json1.html), not by looking at individual bytes of the BLOB.

 (compare [https://sqlite.org/jsonb.html](https://sqlite.org/jsonb.html))

* the plugin might be flaky on returning / injecting json
* hashing / stringify relies on sqlite which might be a good thing so

**Comment by samuelstroschein** [comment]
Notes before starting to solve this with corresponding tasks: 

1. `SELECT json(column)` resolves the ambigious "what is a json" column in auto parsing. 

- [ ] test a query utility that works with `selectAll()` and select the json column

2.  A kysely plugin which creates the snapshot.id based on json content before it's a binary is the way to go to avoid sqlite's internal json format. 
   1. worst case, only the change queue stores snapshots content addressable if a kysely plugin takes too long

- [ ] test kysely plugin to create snapshots ids before it hits sqlite
- [ ] worst case only store snapshots content addressable in change queue 

3. the snapshot.content properties should be sorted to increase de-duplication 

- [ ] add sorting to snapshot content

---

out of scope: 

* using snapshots ids for [LIXDK-114](https://linear.app/opral/issue/LIXDK-114/make-diff-and-change-generation-fault-tolerant). we need to traverse the changes in a version for comparison anyways

**Comment by samuelstroschein** [comment]
query utility has the following downsides: 

* json binary is parsed to a json text but still requires js object parsing (JSON.parse) aka meh
* type chaos between is this a blob or json

it's better to have full parsing plugin that requires no json utility
